x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecycleExecutionState.java

```diff
@@ -310,8 +321,14 @@ public Builder setStepTime(Long stepTime) {
             return this;
         }
 
+        public Builder setIndexOriginationDate(Long originationDate) {
```

Comment:
I think we should approach this slightly differently, rather than storing the origination date in the `LifecycleExecutionState`.

I think it should be a regular index setting (keeping the original creation date in the `LifecycleExecutionState`), then, when we go to use the creation date, we retrieve the setting (if set) and use that rather than the date stored in the execution state. This has the benefit of allowing a user to update the `index.lifecycle.origination_date` setting after the index has already been created, or remove the setting after previously overriding the date (we can fall back to the original creation date).

Reply:
Ah yes, great point